File: mobile/src/main/java/uk/co/appsbystudio/geoshare/utils/ui/NoSwipeViewPager.java
package uk.co.appsbystudio.geoshare.utils.ui;

import android.annotation.SuppressLint;
import android.content.Context;
import android.support.v4.view.ViewPager;
import android.util.AttributeSet;
import android.view.MotionEvent;


public class NoSwipeViewPager extends ViewPager {

    private boolean enabled = false;

    public NoSwipeViewPager(Context context, AttributeSet attributeSet) {
        super(context, attributeSet);
        this.enabled = false;
    }

    @SuppressLint("ClickableViewAccessibility")
    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        return this.enabled && super.onTouchEvent(ev);
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        return this.enabled && super.onInterceptTouchEvent(ev);
    }

    public void setPagingEnabled() {
        this.enabled = false;
    }
}

File: mobile/src/main/java/uk/co/appsbystudio/geoshare/utils/ui/NoSwipeViewPager.kt
package uk.co.appsbystudio.geoshare.utils.ui

import android.annotation.SuppressLint
import android.content.Context
import androidx.viewpager.widget.ViewPager
import android.util.AttributeSet
import android.view.MotionEvent


class NoSwipeViewPager(context: Context, attributeSet: AttributeSet) : ViewPager(context, attributeSet) {

    private var paging: Boolean = false

    init {
        this.paging = true
    }

    @SuppressLint("ClickableViewAccessibility")
    override fun onTouchEvent(ev: MotionEvent): Boolean {
        return this.paging && super.onTouchEvent(ev)
    }

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        return this.paging && super.onInterceptTouchEvent(ev)

    }

    fun setPagingEnabled(enabled: Boolean) {
        this.paging = enabled
    }
}
